docs/src/man/reshaping_and_pivoting.md

```diff
@@ -2,18 +2,91 @@
 
 Reshape data from wide to long format using the `stack` function:
 
-```julia
-using DataFrames
-using CSV
-iris = CSV.read(joinpath(Pkg.dir("DataFrames"), "test/data/iris.csv"), DataFrame)
-iris[:id] = 1:size(iris, 1)  # this makes it easier to unstack
-d = stack(iris, 1:4)
+```jldoctest
+julia> using DataFrames, CSV
+
+julia> iris = CSV.read(joinpath(Pkg.dir("DataFrames"), "test/data/iris.csv"));
+
+julia> head(iris)
```

Comment:
If we just call `iris`, don't we get the first and last rows already?

Reply:
Yes, it's just distractingly long. We can probably come up with something better than chaining `head` and `tail` calls though.

Reply:
Maybe we can find a way to set the `displaysize`?

Reply:
I couldn't get our ideas to work on this. setting `ENV["ROWS"]` and `ENV["LINES"]` [from here](https://github.com/JuliaLang/julia/blob/903644385b91ed8d95e5e3a5716c089dd1f1b08a/base/stream.jl#L351-L353) don't appear to have any effect on our show methods. The `IOContext` approach works as long as you explicitly call `show(IOContext, df)`, but I couldn't find a way to set it in a way that would globally change the DataFrame printing. I think we should try and make `ENV["LINES"]` apply to printing (the show + IOContext didn't seem like a useful thing to show in the introductory doctests). It seems that we're already accounting for `ENV["COLUMNS"]` [here](https://github.com/JuliaData/DataFrames.jl/blob/master/src/abstractdataframe/show.jl#L182) & [here](https://github.com/JuliaData/DataFrames.jl/blob/master/src/abstractdataframe/show.jl#L330)

Reply:
Too bad. I think that's because `displaysize(STDOUT)` ends up calling `uv_tty_get_winsize` instead of relying on `ENV`. A possible hack would be to override `Base.show(df::AbstractDataFrame, allcols::Bool = false)` to call something like `show(IOContext(STDOUT, :displaysize=>(10, 80), df, allcols))`.